src/reanimated2/platform-specific/webAnimations.ts

```diff
@@ -0,0 +1,278 @@
+  }
+}
+
+/**
+ *  Modifies default animation by preserving transformations that given element already contains.
+ *
+ * @param animationName Name of the animation to be modified (e.g. `FadeIn`).
+ * @param existingTransform Transform values that element already contains.
+ * @returns Animation parsed to keyframe string.
+ */
+export function createAnimationWithExistingTransform(
+  animationName: string,
+  existingTransform: NonNullable<TransformsStyle['transform']>
+): string {
+  if (!(animationName in Animations)) {
+    return '';
+  }
+
+  const keyframeName = generateRandomKeyframeName();
+  const newAnimationData = structuredClone(AnimationsData[animationName]);
+
+  newAnimationData.name = keyframeName;
+
+  if (typeof existingTransform === 'string') {
+    throw new Error(
+      '[Reanimated] String transform is not currently supported.'
+    );
+  }
+
+  type TransformPropType = (typeof existingTransform)[number];
+
+  const newTransform = existingTransform.map(
+    (transformProp: TransformPropType) => {
+      const newTransformProp: TransformProperties = {};
+      for (const [key, value] of Object.entries(transformProp)) {
+        if (key.includes('translate')) {
+          // @ts-ignore After many trials we decided to ignore this error - it says that we cannot use 'key' to index this object.
+          // Sadly it doesn't go away after using cast `key as keyof TransformProperties`.
+          newTransformProp[key] = `${value}px`;
+        } else {
+          // @ts-ignore same as above.
+          newTransformProp[key] = value;
+        }
+      }
+      return newTransformProp;
+    }
+  );
+
+  for (const timestampProperties of Object.values(newAnimationData.style)) {
+    if (!timestampProperties.transform) {
+      timestampProperties.transform = newTransform;
+    } else {
+      // We insert existing transformations before ours.
+      Array.prototype.unshift.apply(
+        timestampProperties.transform,
+        newTransform
+      );
+    }
+  }
+
+  const keyframe = parseAnimationObjectToKeyframe(newAnimationData);
+
+  const styleTag = document.getElementById(
+    WEB_ANIMATIONS_ID
+  ) as HTMLStyleElement;
+
+  styleTag.sheet?.insertRule(keyframe);
+
+  return keyframeName;
+}
+
+export function generateRandomKeyframeName() {
+  const characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  const length = 50;
+  let keyframeName = '';
+
+  for (let i = 0; i < length; i++) {
+    keyframeName += characters[Math.floor(Math.random() * characters.length)];
+  }
+
+  return keyframeName;
+}
+
+export function getEasingFromConfig(config: ConfigType): string {
+  if (!('easingV' in config)) {
+    return `cubic-bezier(${WebEasings.linear.toString()})`;
+  }
+
+  config = config as CustomConfig;
+
+  const easingName =
+    config.easingV !== undefined && config.easingV.name in WebEasings
+      ? config.easingV.name
+      : 'linear';
+
+  return `cubic-bezier(${WebEasings[easingName].toString()})`;
+}
```

Comment:
As discussed, and the following functions in similar fashion.
```suggestion
export function getEasingFromConfig(config: ConfigType | CustomConfig) {
  if (!('easingV' in config)) {
    return `cubic-bezier(${WebEasings.linear.toString()})`;
  }

  const easingName =
    config.easingV !== undefined &&
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    (config as CustomConfig).easingV!.name in WebEasings
      ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        (config as CustomConfig).easingV!.name
      : 'linear';

  return `cubic-bezier(${WebEasings[easingName].toString()})`;
}
```

Reply:
Done in [cda2f56](https://github.com/software-mansion/react-native-reanimated/pull/4805/commits/cda2f56442e0197ebfcf64bf0df1d1ca5af8baf6)